### ISSUE TITLE
JSONTest unit test is failing.

### ISSUE THREAD

**Issue by tgvaughan** [CompEvol/beast2#918]
This is due to commit 06a638d6 which changed the default scaleFactor input value `ScaleOperator` to 0.75.

At first this sounds unrelated to JSONTest, which deserializes and then serializes a model from and to the JSON format.  However, the model includes a ScaleOperator and the value 0.75  happens to coincide with the value used in the model, meaning that the JSON serializer `JSONProducer` now omits this input from the JSON string it produces, causing it to differ from the original.

Just an aside: I'm worried that we're beginning to ignore unit test failures.  This might be because of the noise produced by the BEAUti gui testing.  If so, we might need to think hard about either (a) fixing the problematic tests (making them pass _deterministically_), (b) substituting them for functionally equivalent but more reliable tests, or (c) removing/disabling them.

Whatever the reason, ignoring even trivial failing tests can lead to big problems, as (a) it's more difficult to notice when an important test fails if other tests are already failing, and (b) it's always going to cost someone way more time and energy to diagnose a problem that was created ages ago than it would have been to find the cause immediately after the problem was introduced. (Although of course it's always fun to solve these kinds of mysteries.. :-) )

**Comment by alexeid** [comment]
Thanks for fixing this! I think we should carve BEAUti tests out, since they seem to cause the biggest problem. I believe Remco plans to make BEAUti a separate package in due course.

**Comment by walterxie** [comment]
Perhaps, everything related to java swing (GUI) could be separated to a package? So, the code to build `beast.jar` will be "swing-free".

**Comment by tgvaughan** [comment]
No worries!  It'd be a shame to completely remove the beauti tests, but we certainly shouldn't risk missing critical bugs in the core because they're unreliable.

I should mention that part of the reason I got worked up about this was that today I also discovered that BDSKY has been failing a unit test since January 2019 (details [here](https://github.com/BEAST2-Dev/bdsky/issues/28)).  I'm as much at fault as anybody, since I've made commits to that repository in the meantime and clearly didn't run the test suite.  But if automatic testing had been working there we would have caught that straight away, rather than 18 months and three releases later.
